### PR TITLE
Allow decrease reclaimable pods to 0 for suspended job

### LINKED ISSUE #1215: When a job resets reclaimable pods when evicted, it gets stuck due to Workload validation
If the workload is preempted, will ReclaimablePod be set to nil?
image
```bash
{"level":"error","ts":"2023-10-16T12:47:55.770288657Z","caller":"controller/controller.go:324","msg":"Reconciler error","controller":"job","controllerGroup":"batch.volcano.sh","controllerKind":"Job","Job":{"name":"hobot-job-xxx","namespace":"cpu-preempt"},"namespace":"cpu-preempt","name":"hobot-job-xxx","reconcileID":"f4abf36a-f5f2-4ec0-bbae-31b49cfef394","error":"admission webhook \"[vworkload.kb.io](http://vworkload.kb.io/)\" denied the request: status.reclaimablePods[main]: Required value: cannot be removed","stacktrace":"[sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/gopath/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:324\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/gopath/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/gopath/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226](http://sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler/n/t/gopath/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:324/nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem/n/t/gopath/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265/nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2/n/t/gopath/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226)"}
```
Is this our normal logic, or am I using it incorrectly, which would cause the webhook checksum to fail for the departure workload.

### PR DESCRIPTION
**Pull request by yaroslava-serdiuk**
#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes:
Fixes #1215

#### Does this PR introduce a user-facing change?
```release-note
Allow decrease reclaimable pods to 0 for suspended job
```

### CODE REVIEW

File: pkg/webhooks/workload_webhook.go [hunk 1]
```diff
@@ -388,14 +388,17 @@ func validateReclaimablePodsUpdate(newObj, oldObj *kueue.Workload, basePath *fie
 	for i := range newObj.Status.ReclaimablePods {
 		newCount := &newObj.Status.ReclaimablePods[i]
 		newNames.Insert(newCount.Name)
+		if !workload.HasQuotaReservation(newObj) && newCount.Count == 0{
```
Comment (mimowo) [inline pkg/webhooks/workload_webhook.go:391]:
```suggestion
		if !workload.HasQuotaReservation(newObj) && newCount.Count == 0 {
```

**Review by mimowo — COMMENTED** [review]
/lgtm
/assign @alculquicondor

**Review by alculquicondor — COMMENTED** [review]
/approve

**Review by tenzen-y — COMMENTED** [review]
/lgtm

**Comment by tenzen-y** [comment]
> /kind api-change

@alculquicondor As I can see, we change only webhooks. Is changing webhooks included in `kind/api-change`?

**Comment by alculquicondor** [comment]
Yeah, a change in the validation is an API change

**Pull request merged**